unit/util/interval/multiply.cpp

```diff
@@ -0,0 +1,144 @@
+/*******************************************************************\
+ Module: Unit tests for variable/sensitivity/abstract_object::merge
+ Author: DiffBlue Limited
+\*******************************************************************/
+
+#include <testing-utils/use_catch.h>
+
+#include <util/arith_tools.h>
+#include <util/interval.h>
+#include <util/std_expr.h>
+#include <util/std_types.h>
+#include <util/symbol_table.h>
+
+#define V(X) (bvrep2integer(X.get(ID_value).c_str(), 32, true))
+#define V_(X) (bvrep2integer(X.c_str(), 32, true))
+#define CEV(X) (from_integer(mp_integer(X), signedbv_typet(32)))
+
+SCENARIO("multiply interval domain", "[core][analyses][interval][multiply]")
+{
+  GIVEN("A selection of constant_exprts in a std::vector and map")
+  {
+    symbol_tablet symbol_table;
+    namespacet ns(symbol_table);
+
+    WHEN("Both are positive [2,5]*[7,11]")
+    {
+      constant_interval_exprt a(CEV(2), CEV(5));
+      constant_interval_exprt b(CEV(7), CEV(11));
+
+      constant_interval_exprt result = a.multiply(b);
+
+      THEN("Domain is consistent")
+      {
+        CHECK(V(a.get_lower()) == 2);
+        CHECK(V(a.get_upper()) == 5);
+        CHECK(V(b.get_lower()) == 7);
+        CHECK(V(b.get_upper()) == 11);
+      }
+
+      CAPTURE(result);
+
+      THEN("The result is [14, 55]")
+      {
+        CHECK(V(result.get_lower()) == 14);
+        CHECK(V(result.get_upper()) == 55);
+      }
+    }
+
+    WHEN("One is entirely negative [-2,-5]*[7,11]")
+    {
+      constant_interval_exprt a(CEV(-5), CEV(-2));
+      constant_interval_exprt b(CEV(7), CEV(11));
+
+      constant_interval_exprt result = a.multiply(b);
+
+      THEN("Domain is consistent")
+      {
+        CHECK(V(a.get_lower()) == -5);
```

Comment:
:no_entry_sign: I'm pretty sure CHECK here means the test won't fail if this is wrong, should be REQUIRE

Reply:
This is wrong - just allows all checks to be tested, and is probably _more_ correct as the results are indepedent. Certainly doesn't need to be fixed.